verification_tokens/decorators.py

```diff
@@ -0,0 +1,89 @@
+from functools import wraps
+from typing import Callable, Optional
+
+from django.core.exceptions import PermissionDenied
+
+from common.utils import context
+from verification_tokens.models import VerificationToken
+
+
+def user_from_auth_verification_token(
+    get_token: Callable[[dict], Optional[str]], use_only_when_first_denied: bool = False
+):
+    """Sets the related auth token user to the context
+    (in place of the logged in user; ``context.user``),
+    if the wrapped function throws a permission denied error and then retries
+    the wrapped function.
```

Comment:
This is now out-of-date with the function implementation as the function can now be made to function differently based on the `use_only_when_first_denied` parameter.